### ISSUE TITLE
Update Sample IDs when concatenating ODM objects

### ISSUE THREAD

**Issue by NauticalMile64** [DeGrootResearchGroup/pyODM#7]
Sample IDs in datasets are normally sequential starting from 1, this needs to be fixed to make sampleIDs unique in concatenated objects.

**Comment by nav-mohan** [comment]
I dont quite understand this issue. The `sampleID` column in the **Sample.csv** and **SiteMeasure.csv** are alpha-numeric values like `THC_20210401`, `VNL_20220130`. They are a combination of the `siteID` and the `dateTimeStart`. From what I can see, the `sampleID` column is not sequential in a dataset. It's not sorted by `dateTime`, nor is it sorted by `siteID`. 
![image](https://github.com/user-attachments/assets/5b909690-53c1-4e83-ad83-e873373d70eb)

Are you perhaps referring to a different "Sample ID"?

I ran a quick test. I took two slices of the CSV dataset in the **example** folder. 
**dataset-1** was composed of the first 10 rows of the **Sample.csv**, **SiteMeasure.csv**
**dataset-2** was composed of the last 10 rows of the **Sample.csv**, **SiteMeasure.csv**
I then combined them and exported them. 
The exported **Sample.csv** had 20 rows because both datasets contributed unique rows.
The exported **SiteMeasure.csv** had 14 rows - 6 rows were dropped because they'd been sliced out from the **Sample.csv**.

**Comment by nav-mohan** [comment]
Sorry, I think I see what you mean by sequential - It is sorted by `dateTime` first and then sorted by `siteID`. Is that correct?

**Comment by ctdegroot** [comment]
Good point @nav-mohan. In the sample dataset that comes from PHAC they have obviously done a better job of generative sampleIDs that are sure to be independent. This issue stems from the Ontario WSI dataset which can be found here: https://github.com/OntarioWastewaterSurveillanceConsortium/sars-cov-2-data. 

For our lab (Western) we just used simple numeric sampleIDs, which I assumed is what other labs were doing as well. However, looking through the dataset, I see that each lab has invented their own nomenclature. Although the issue is not the one that I was originally thinking, I believe there is the possibility that when concatenating different datasets, there could be a risk that there are duplicate sampleIDs coming from different files.

So... I think that what we need to do there is check that there are no duplicate sampleIDs when concatenating ODM objects, since any duplication would be by chance, not because they refer to the same sample. If there is duplication, then we can just append something to the end of the overlapping sampleID, e.g. "_0", "_1", etc if there are multiple duplicates.

**Comment by nav-mohan** [comment]
Thank you for pointing me to those alternate datasets. That helped with understanding the case. 
Yes, I see what you mean - each lab has it's own arbitrary naming convention for the `sampleID` column. This could potentially lead to two labs coincidentally assigning identical `sampleID` for two records, even though those two records are not necessarily duplicates.  

If my understanding of this issue is correct then before implementing a fix, we should also have a strategy on how to identify truly duplicate records. I see that when we concat two tables [we drop duplicates](https://github.com/DeGrootResearchGroup/pyODM/blob/622642ab3bef1ae259934161298ce6f9d15b9825/pyodm/odm.py#L304) if the two records have identical values for the first column. In the case of the **Samples.csv** table, the first column is `sampleID`. So, if we were to allow duplicate values for `sampleID` by modifying the second occurrence instead of dropping it, then we should have an alternate strategy for identifying duplicates. Perhaps, if ALL the other column values are identical?

On a related note, are there any other columns that are also arbitrarily named? Such as `siteID`?

**Comment by ctdegroot** [comment]
I think that we should only drop duplicates if all of the other column values are identical. This would eliminate the possibility where the sampleID is the same by chance when combining datasets. If all of the values are identical then I think it is safe to assume it is a duplicate record.

For the dataset that I referenced above we can be sure that the siteID will not be duplicated, since these were chosen to avoid overlap. However, I suppose that we cannot guarantee this is the case for other datasets, so it is probably a good idea to do a check on this. For the siteID there are geographic coordinates that can help us determine if the sites are the same or not.

**Comment by nav-mohan** [comment]
I've [opened a PR](https://github.com/DeGrootResearchGroup/pyODM/pull/21) from my fork.
